### ISSUE TITLE
missing text after equation blogger in safari not Firefox/chrome

### ISSUE THREAD

**Issue by makeyourownalgorithmicart** [mathjax/MathJax#2005]
The following blogger text leads to the text after the second equation to be missing in Safari, but not in Firefox or Chrome.

```
Here's a simple example:

$$ \sum_{n=1}^{\infty} {n} =  1 + 2 + 3 + 4 + \cdots $$

As we keep adding these ever increasing numbers, the sum gets larger and larger and larger. The sum of this series is infinity.

What about this one:

$$ \sum_{n=1}^{\infty} {n} =  1 + 2 + 3 + 4 + \cdots $$

saaa

bbb

ccc

ddd
```

The mathjax library is included as per usual in the html:

`<script type="text/x-mathjax-config">   MathJax.Hub.Config({tex2jax: {inlineMath: [['$','$'], ['\\(','\\)']]}}); </script> <script src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"> </script>`

I've been using mathjax in blogger for years, and this seems to be a new/recent problem. 

All software is latest - MacOS, browsers and mathjax.

Here is the output from Safari:
<img width="732" alt="safari" src="https://user-images.githubusercontent.com/29485854/41052087-d2b63f70-69af-11e8-9fb9-30da008665f6.png">

Here's Firefox:
<img width="774" alt="firefox" src="https://user-images.githubusercontent.com/29485854/41052107-d60beeea-69af-11e8-91a9-ec2083036852.png">

And Chrome:
<img width="749" alt="chrome" src="https://user-images.githubusercontent.com/29485854/41052111-d8abe754-69af-11e8-9aff-11f05bd16b30.png">

**Comment by dpvc** [comment]
Can you give a link to the blog page itself?  I don't think we can diagnose this without seeing the actual page.

**Comment by makeyourownalgorithmicart** [comment]
here is a blog that works ok with firefox and chrome but not safari on macosx - latest OS and latest browsers

[http://makeyourownalgorithmicart.blogspot.com/2018/06/visualising-riemann-zeta-function.html](http://makeyourownalgorithmicart.blogspot.com/2018/06/visualising-riemann-zeta-function.html)

**Comment by dpvc** [comment]
Thanks for the link.  I can reproduce the result.  Very strange.  The DOM looks correct, but the elements are not being shown.  I don't know why.  But there are two ways that I found to avoid the problem.

You could turn off the fat preview using

```
MathJax.Hub.Config({
  'fast-preview': {disabled: true}
});
```

or you could switch to the SVG output renderer.  Both of these work properly for me.

**Comment by makeyourownalgorithmicart** [comment]
Thanks @dpvc disabling fast-preview worked. 
Good luck with finding the error.

And thanks for a great bit of software - incredibly useful!

**Comment by dpvc** [comment]
Additional examples listed at [physics forum](https://www.physicsforums.com/threads/problem-opening-certain-threads-in-safari.943990/).  Disabling fast-preview or switching to SVG works there.  Still not sure what is happening.

**Comment by bbirkett** [comment]
Here's [another example](https://www.matheno.com/safari-crasher.html) in case it's useful to you: fine in Chrome and Firefox, but won't fully load in Safari.  A user reported a particular page on our site was crashing his Mac-Safari, and we isolated this small block as the issue.  Disabling fast-preview or switching to SVG does fix the problem, but comes at other costs so we're asking users not to use Mac-Safari for now.

Thanks for creating and maintaining MathJax, for which I have an ever-greater appreciation, and for working through issues like this when they arise!

**Comment by dpvc** [comment]
@bbirkett, thanks for the additional link, especially for providing a minimal example of the problem.

**Comment by dpvc** [comment]
Here is an even more minimal example:

```
<!DOCTYPE html>
<html>
<head>
<title>Safari Crasher</title>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/latest.js?config=TeX-AMS_HTML-full"></script>
</head>
<body>
<div>
<hr>
\[x\]
<em>\(x\)</em>
</div>
</body>
</html>
```

Removing any of the elements in the body (the `<div>`, the `<hr>`, or the `<em>`) will allow the page to process normally.  Disabling the fast preview also allows the page to process.

**Comment by dpvc** [comment]
Here is a minimal example for the missing text:

```
<!DOCTYPE html>
<html>
<head>
<title>Safari Missing Text</title>
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  messageStyle: "none"
});
</script>
<script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/MathJax.js?config=TeX-AMS_CHTML-full"></script>
</head>

<body>
$$x$$
$$x$$
Missing text

<div></div>

</body>
</html>
```
Note that the final `<div>` and two displayed equations are necessary, as is the `messageStyle` setting.

**Comment by dpvc** [comment]
OK, it turns out that both of these problems are triggered when MathJax sets they preview to have `display: none` styling.  This occurs at [line 772](https://github.com/mathjax/MathJax/blob/master/unpacked/jax/output/HTML-CSS/jax.js#L772) of the HTML-CSS output, [line 614](https://github.com/mathjax/MathJax/blob/master/unpacked/jax/output/CommonHTML/jax.js#L614) of the CommonHTML output, and [line 405](https://github.com/mathjax/MathJax/blob/master/unpacked/jax/output/SVG/jax.js#L405) of the SVG output (though the latter doesn't seem to trigger the problems).  Since the preview element's contents is cleared on the line before, there is really no need to set `display: none`, so this can be dropped.

For now, you can add

```
<script type="text/x-mathjax-config">
MathJax.Hub.Register.StartupHook("HTML-CSS Jax Ready", function () {
  var HTMLCSS = MathJax.OutputJax["HTML-CSS"];
  var postTranslate = HTMLCSS.postTranslate;
  HTMLCSS.Augment({
    postTranslate: function (state, partial) {
      var scripts = state.jax[this.id];
      for (var i = state.HTMLCSSlast, m = state.HTMLCSSeqn; i < m; i++) {
        var script = scripts[i];
        if (script && script.parentNode && script.MathJax.elementJax) {
          if (script.MathJax.preview) {
            script.MathJax.preview.innerHTML = "";
            delete script.MathJax.preview;
          }
        }
      }
      postTranslate.call(this, state, partial);
    }
  });
});
MathJax.Hub.Register.StartupHook("CommonHTML Jax Ready", function () {
  var CHTML = MathJax.OutputJax.CommonHTML;
  var postTranslate = CHTML.postTranslate;
  CHTML.Augment({
    postTranslate: function (state) {
      var scripts = state.jax[this.id];
      for (var i = state.CHTMLlast, m = state.CHTMLeqn; i < m; i++) {
        var script = scripts[i];
        if (script && script.MathJax.elementJax) {
          var data = script.MathJax.elementJax.CHTML;
          if (data.preview) {
            data.preview.innerHTML = "";
            delete data.preview;
          }
        }
      }
      postTranslate.call(this, state);
    }
  });
});
</script>
```

to your pages just *before* the script that loads `MathJax.js` in order to work around the problem until the next version of MathJax (that will include the fix) is released.

**Comment by bbirkett** [comment]
@dpvc, thank you for tracking this down and for the fix!

**Comment by dpvc** [comment]
You're welcome.  If you try the patch, will you let us know if it works for you?

**Comment by bbirkett** [comment]
@dpvc  Yes, the fix worked and the page no longer crashes --- thanks!  There are still the errant line breaks in WebKit as others have referenced (#1982), but I trust that you've addressed those in #2006 and so they'll be fixed in the next release.  Thank you again, VERY much, for addressing this so quickly.   Really: very much appreciated.

**Comment by dpvc** [comment]
==> In testsuite
MathMLToDisplay/issue2005.html
